File: 0539 Minimum Time Difference/0539.cpp
class Solution {
public:
    int mins(string time){
        int h = stoi(time.substr(0,2));
        int m = stoi(time.substr(3,2));
        return h * 60 + m;
    }

    int findMinDifference(vector<string>& times) {
        sort(times.begin(), times.end());
        int ans = 24 * 60 - mins(times.back()) + mins(times.front());
        for (int i = 1; i < times.size(); i++){
            ans = min(ans, mins(times[i]) - mins(times[i-1]));
        }
        return ans;
    }
};

File: 0539 Minimum Time Difference/0539.py
class Solution:
    def findMinDifference(self, times: List[str]) -> int:
        def mins(time):
            h, m = [int(x) for x in time.split(':')]
            return h * 60 + m
            
        times.sort()
        ans = 24 * 60 - mins(times[-1]) + mins(times[0])
        for i in range(1, len(times)):
            ans = min(ans, mins(times[i]) - mins(times[i-1]))
        return ans
